include/deal.II/lac/scalapack.h

```diff
@@ -0,0 +1,478 @@
+ *      |   .  -6.0 -2.0  0.0  -2.0  |  -2.0 -4.0  0.0 -4.0
+ *  0   |   .    .  -6.0 -2.0   6.0  |  -2.0  0.0  2.0  0.0
+ *      |   .    .    .  -6.0   2.0  |   2.0  0.0  2.0  0.0
+ *      |   .    .    .    .  -16.0  |    .    .    .    .
+ * -----|----------------------------|----------------------
+ *      |   .    .    .    .    0.0  |  -8.0 -4.0  0.0 -2.0
+ *      |   .    .    .    .   -6.0  |    .  -6.0  0.0 -4.0
+ *  1   |   .    .    .    .    0.0  |    .    .  -4.0  0.0
+ *      |   .    .    .    .   -4.0  |    .    .    .  -4.0
+ * @endcode
+ *
+ *
+ * Here is a strong scaling example of ScaLAPACKMatrix::invert()
+ * on up to 5 nodes each composed of two Intel Xeon 2660v2 IvyBridge sockets
+ * 2.20GHz, 10 cores/socket. Calculate are performed on square processor
+ * grids 1x1, 2x2, 3x3, 4x4, 5x5, 6x6, 7x7, 8x8, 9x9, 10x10.
+ *
+ * @image html scalapack_invert.png
+ *
+ * @ingroup Matrix1
+ * @author Denis Davydov, Benjamin Brands, 2017
+ */
+template <typename NumberType>
+class ScaLAPACKMatrix : protected TransposeTable<NumberType>
+{
+public:
+
+  /**
+    * Declare type for container size.
+    */
+  typedef unsigned int size_type;
+
+  /**
+   * Constructor for a rectangular matrix with rows and columns provided in
+   * @p sizes, distributed using the grid @p process_grid .
+   *
+   * The choice of the block size @p block_sizes is a compromise between a large
+   * enough sizes for efficient local BLAS and small enough to get
+   * good parallel load balance.
+   */
+  ScaLAPACKMatrix(const std::pair<size_type,size_type> &sizes,
+                  std::shared_ptr<ProcessGrid> process_grid,
+                  const std::pair<size_type,size_type> &block_sizes = std::make_pair(32,32),
+                  const LAPACKSupport::Property property = LAPACKSupport::Property::general);
+
+  /**
+   * Constructor for a square matrix of size @p size, distributed
+   * using the process grid in @p process_grid.
+   *
+   * The choice of the block size @p block_size is a compromize between a large
+   * enough sizes for efficient local BLAS and small enough get
+   * good parallel load balance.
+   */
+  ScaLAPACKMatrix(const size_type size,
+                  std::shared_ptr<ProcessGrid> process_grid,
+                  const unsigned int block_size = 32);
+
+  /**
+   * Destructor
+   */
+  virtual ~ScaLAPACKMatrix();
+
+  /**
+   * Assignment operator from a regular FullMatrix.
+   */
+  ScaLAPACKMatrix<NumberType> &
+  operator = (const FullMatrix<NumberType> &);
+
+  /**
+   * Copy the content of the distributed matrix into @p matrix.
+   */
+  void copy_to (FullMatrix<NumberType> &matrix) const;
+
+  /**
+   * Compute the Cholesky factorization of the matrix using ScaLAPACK function pXpotrf.
+   */
+  void compute_cholesky_factorization ();
+
+  /**
+   * Invert the matrix by first computing Cholesky factorization and then
+   * building the actual inverse using pXpotri.
+   */
+  void invert();
```

Comment:
does this replace the contents of ``*this`` by the inverse? If yes, please document that.